TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/AbstractNodeData.php

```diff
@@ -281,7 +281,7 @@ public function getProperties()
         }
 
         $properties = array();
-        foreach (array_keys($this->properties) as $propertyName) {
+        foreach ($this->properties as $propertyName => $propertyValue) {
```

Comment:
left it like this. I think it's more obvious what's happening ("plain" foreach).